src/vcpkg/commands.format-manifest.cpp

```diff
@@ -197,73 +282,62 @@ namespace vcpkg::Commands::FormatManifest
         auto parsed_args = args.parse_arguments(COMMAND_STRUCTURE);
 
         auto& fs = paths.get_filesystem();
-        bool has_error = false;
 
         const bool format_all = Util::Sets::contains(parsed_args.switches, OPTION_ALL);
         const bool convert_control = Util::Sets::contains(parsed_args.switches, OPTION_CONVERT_CONTROL);
 
         if (!format_all && convert_control)
         {
-            System::print2(System::Color::warning, R"(format-manifest was passed '--convert-control' without '--all'.
-    This doesn't do anything:
-    we will automatically convert all control files passed explicitly.)");
+            System::print2(System::Color::warning, R"(Warning: '--convert-control' has no effect without '--all'.
+    Explicit paths to CONTROL files, or names of ports using CONTROL files
+    are always converted.)");
         }
 
         if (!format_all && args.command_arguments.empty())
         {
-            Checks::exit_with_message(
-                VCPKG_LINE_INFO,
-                "No files to format; please pass either --all, or the explicit files to format or convert.");
+            Checks::exit_with_message(VCPKG_LINE_INFO, R"(Error: No targets to format. Please pass --all,
+    or a list of port names, CONTROL files, and/or manifests to format or convert.)");
         }
 
+        std::vector<std::string> errors;
         std::vector<ToWrite> to_write;
-
-        const auto add_file = [&to_write, &has_error](Optional<ToWrite>&& opt) {
-            if (auto t = opt.get())
-                to_write.push_back(std::move(*t));
-            else
-                has_error = true;
-        };
-
         for (const auto& arg : args.command_arguments)
         {
-            auto path = fs::u8path(arg);
-            if (path.is_relative())
+            auto maybe_resolved =
+                resolve_format_manifest_input(arg, paths.original_cwd, paths.builtin_ports_directory(), fs);
```

Comment:
We should restrict the "by-name" lookups to only occur if the original_cwd is within the vcpkg root. This reserves future design.

Unsorted notes: (sorry!)
```
~$ vcpkg edit <<<<<
~$ vcpkg x-add-version <boost>
/my/registry$ vcpkg format-manifest boost
/vcpkg$ vcpkg format-manifest boost 
/vcpkg/vcpkg-configuration.json -> "boost": "/my/registry"
```